rewriter/Minitest.cc

```diff
@@ -276,21 +316,66 @@ ast::ExpressionPtr runUnderEach(core::MutableContext ctx, core::NameRef eachName
             // add back any moved constants
             return constantMover.addConstantsToExpression(send->loc, move(method));
-        } else if (send->fun == core::Names::describe() && send->numPosArgs() == 1 && correctBlockArity) {
+        } else if ((isDescribeOrSimilar(send->fun) || isSharedExamples(send->fun)) && send->numPosArgs() == 1 &&
+                   correctBlockArity) {
             return prepareTestEachBody(ctx, eachName, std::move(send->block()->body), args, destructuringStmts,
                                        iteratee,
                                        /* insideDescribe */ true);
-        } else if (insideDescribe && send->fun == core::Names::let() && send->numPosArgs() == 1 && correctBlockArity &&
-                   ast::isa_tree<ast::Literal>(send->getPosArg(0))) {
-            auto argLiteral = ast::cast_tree_nonnull<ast::Literal>(send->getPosArg(0));
-            if (argLiteral.isName()) {
-                auto declLoc = send->loc.copyWithZeroLength().join(argLiteral.loc);
-                auto methodName = argLiteral.asName();
-                return ast::MK::SyntheticMethod0(send->loc, declLoc, methodName, std::move(send->block()->body));
+        } else if (insideDescribe && send->fun == core::Names::itBehavesLike() && send->numPosArgs() == 1) {
+            // Handle it_behaves_like within test_each blocks
+            auto &arg = send->getPosArg(0);
+            auto argString = to_s(ctx, arg);
+            // Create an include statement that includes the shared_examples companion module
+            auto moduleName = makeSharedExamplesModuleConstantLocal(ctx, arg.loc(), argString);
+            // Create an include statement to include the shared_examples companion module
+            return ast::MK::Send1(send->loc, ast::MK::Self(send->loc), core::Names::include(), send->funLoc,
+                                  std::move(moduleName));
+        } else if (insideDescribe &&
+                   ((send->fun == core::Names::let() && send->numPosArgs() == 1) ||
+                    (send->fun == core::Names::letBang() && send->numPosArgs() == 1) ||
+                    (send->fun == core::Names::subject() && send->numPosArgs() <= 1)) &&
+                   correctBlockArity && (send->numPosArgs() == 0 || ast::isa_tree<ast::Literal>(send->getPosArg(0)))) {
+            core::NameRef methodName;
+            core::LocOffsets declLoc;
+
+            if (send->numPosArgs() == 1) {
+                auto argLiteral = ast::cast_tree_nonnull<ast::Literal>(send->getPosArg(0));
+                if (argLiteral.isName()) {
+                    declLoc = send->loc.copyWithZeroLength().join(argLiteral.loc);
+                    methodName = argLiteral.asName();
+                }
+            } else {
+                declLoc = send->loc.copyWithZeroLength().join(send->funLoc);
+                methodName = core::Names::subject();
```

Comment:
This code does not unconditionally set `methodName` before the call to `MK::SyntheticMethod0` is reached, which means that if you have a `let` with a non-`Symbol` arg0 inside a describe block inside a test_each block, Sorbet will crash.